brooklyn-server/server-cli/src/main/java/org/apache/brooklyn/cli/ItemLister.java

```diff
@@ -98,88 +115,150 @@ public Void call() throws Exception {
+                                    param.getLabel(), null, MutableMap.<String,URI>of());
+                            config.add(entityConfigSummary);
+                        }
+                        itemDescriptor.put("config", config);
+                        if (item.getCatalogItemType().name().equals("ENTITY")) {
+                            entities.add(itemDescriptor);
+                        } else if (item.getCatalogItemType().name().equals("POLICY")) {
+                            policies.add(itemDescriptor);
+                        } else if (item.getCatalogItemType().name().equals("LOCATION")) {
+                            locations.add(itemDescriptor);
+                        }
+                    }
+                    result = ImmutableMap.<String, Object>builder()
+                            .put("entities", entities)
+                            .put("policies", policies)
+                            .put("enrichers", enrichers)
+                            .put("locations", locations)
+                            .put("locationResolvers", locationResolvers)
+                            .build();
+
+                    jsonList.add(toJson(result));
                 }
-                
-                List<Map<String, Object>> locations = (List<Map<String, Object>>) result.get("locations");
-                String locationTemplateHtml = resourceUtils.getResourceAsString(Urls.mergePaths(BASE_TEMPLATES, "location.html"));
-                for (Map<String, Object> location : locations) {
-                    String type = (String) location.get("type");
-                    String locationHtml = TemplateProcessor.processTemplateContents(locationTemplateHtml, ImmutableMap.of("type", type));
-                    Files.write(locationHtml, new File(Os.mergePaths(outputFolder, "locations", type + ".html")), Charsets.UTF_8);
+            }
+
+            for (String json: jsonList) {
+
+                if (outputFolder == null) {
+                    System.out.println(json);
+                } else {
+                    LOG.info("Outputting item list (size "+itemCount+") to " + outputFolder);
+                    String outputPath = Os.mergePaths(outputFolder, "index.html");
+                    String parentDir = (new File(outputPath).getParentFile()).getAbsolutePath();
+                    mkdir(parentDir, "entities");
+                    mkdir(parentDir, "policies");
+                    mkdir(parentDir, "enrichers");
+                    mkdir(parentDir, "locations");
+                    mkdir(parentDir, "locationResolvers"); //TODO nothing written here yet...
+
+                    mkdir(parentDir, "style");
+                    mkdir(Os.mergePaths(parentDir, "style"), "js");
+                    mkdir(Os.mergePaths(parentDir, "style", "js"), "catalog");
+
+                    Files.write("var items = " + json, new File(Os.mergePaths(outputFolder, "items.js")), Charsets.UTF_8);
```

Comment:
I don't think it matters in this case @neykov as a blueprint author will use the `br list-objects` command to generate this file for his YAML blueprint then copy it onto his blueprint git repository